src/VisualStudio/Core/Impl/Options/GridOptionPreviewControl.xaml.cs

```diff
@@ -93,5 +106,229 @@ internal override void Close()
             }
         }
+
+        internal void Generate_Editorconfig(object sender, System.Windows.RoutedEventArgs e)
+        {
+            var editorconfig = new StringBuilder();
+            var optionSet = this.ViewModel.ApplyChangedOptions(this.OptionService.GetOptions());
+
+            GenerateEditorconfig_CoreSettings(optionSet, _language, editorconfig);
+            GenerateEditorconfig_DotNetSettings(optionSet, _language, editorconfig);
+
+            // Generate language specific settings
+            _getCurrentEditorConfigOptions(optionSet, editorconfig);
+
+            var sfd = new System.Windows.Forms.SaveFileDialog
+            {
+                Filter = "All files (*.*)|",
+                FileName = ".editorconfig",
+                Title = "Save .editorconfig file"
+            };
+            if (sfd.ShowDialog() == System.Windows.Forms.DialogResult.OK)
+            {
+                IOUtilities.PerformIO(() =>
+                {
+                    var path = sfd.FileName;
+                    using (var sw = new StreamWriter(File.Create(path)))
+                    {
+                        sw.Write(editorconfig);
+                        sw.Close();
+                    }
+                });
+            }
+        }
+
+        internal static void GenerateEditorconfig_CoreSettings(OptionSet optionSet, string language, StringBuilder editorconfig)
+        {
+            // Core EditorConfig Options
+            editorconfig.AppendLine("# Core EditorConfig Options");
+            editorconfig.AppendLine("# Uncomment the line below if you don’t want to inherit parent .editorconfig settings or if this is your solution directory.");
+            editorconfig.AppendLine("# root = true");
+
+            editorconfig.AppendLine();
+            if (language == LanguageNames.CSharp)
+            {
+                editorconfig.AppendLine("# C# files");
+                editorconfig.AppendLine("[*.cs]");
+            }
+            else if (language == LanguageNames.VisualBasic)
+            {
+                editorconfig.AppendLine("# Basic files");
+                editorconfig.AppendLine("[*.vb]");
+            }
+
+            // indent_style
+            CoreCodeStyleOptions_GenerateEditorconfig(optionSet, FormattingOptions.UseTabs, language, editorconfig);
+            // indent_size
+            CoreCodeStyleOptions_GenerateEditorconfig(optionSet, FormattingOptions.IndentationSize, language, editorconfig);
+            // insert_final_newline
+            CoreCodeStyleOptions_GenerateEditorconfig(optionSet, FormattingOptions.InsertFinalNewLine, editorconfig);
+        }
+
+        private static void CoreCodeStyleOptions_GenerateEditorconfig(OptionSet optionSet, PerLanguageOption<bool> option, string language, StringBuilder editorconfig)
+        {
+            editorconfig.Append(((EditorConfigStorageLocation<bool>)option.StorageLocations.OfType<IEditorConfigStorageLocation>().SingleOrDefault()).KeyName);
```

Comment:
As we discussed offline, this probably needs a null check and bail out. Similar change is needed in all other methods.